gallery/how_to/deploy_models/deploy_model_on_adreno.py

```diff
@@ -0,0 +1,298 @@
+# machine to the remote Adreno device.
+
+rpc_tracker_host = os.environ.get("TVM_TRACKER_HOST")
+rpc_tracker_port = int(os.environ.get("TVM_TRACKER_PORT"))
+key="android"
+
+tracker = rpc.connect_tracker(rpc_tracker_host, rpc_tracker_port)
+remote = tracker.request(
+	key, priority=0, session_timeout=600000)
+
+temp = utils.tempdir()
+dso_binary = "dev_lib_cl.so"
+dso_binary_path = temp.relpath(dso_binary)
+lib.export_library(dso_binary_path, ndk.create_shared)
+remote_path = "/data/local/tmp/" + dso_binary
+remote.upload(dso_binary_path)
+rlib = remote.load_module(dso_binary)
+ctx = remote.cl(0)
+m = graph_executor.GraphModule(rlib["default"](ctx))
+
+######################################################################
+# Run inference on Adreno
+# -----------------------
+m.set_input(input_name, tvm.nd.array(img.astype("float32")))
+m.run()
+tvm_output = m.get_output(0)
+
+######################################################################
+# Get predictions and performance statistic
+# -----------------------------------------
+# This piece of code displays the top-1 and top-5 predictions, as
+# well as provides information about the model's performance
+from  os.path  import  join, isfile
+from  matplotlib  import  pyplot  as  plt
+from  tvm.contrib  import  download
+
+# Download ImageNet categories
+categ_url = "https://github.com/uwsampl/web-data/raw/main/vta/models/"
+categ_fn = "synset.txt"
+download.download(join(categ_url, categ_fn), categ_fn)
+synset = eval(open(categ_fn).read())
+
+top_categories = np.argsort(tvm_output.asnumpy()[0])
+top5 = np.flip(top_categories, axis=0)[:5]
+
+# Report top-1 classification result
+print("Top-1 id: {}, class name: {}".format(top5[1-1], synset[top5[1-1]]))
+
+# Report top-5 classification results
+print("\nTop5 predictions: \n")
+print("\t#1:", synset[top5[1-1]])
+print("\t#2:", synset[top5[2-1]])
+print("\t#3:", synset[top5[3-1]])
+print("\t#4:", synset[top5[4-1]])
+print("\t#5:", synset[top5[5-1]])
+print("\t", top5)
+ImageNetClassifier = False
+for  k  in  top_categories[-5:]:
+	if  "cat"  in  synset[k]:
+		ImageNetClassifier = True
+assert  ImageNetClassifier, "Failed ImageNet classifier validation check"
+
+print("Evaluate inference time cost...")
+print(m.benchmark(ctx, number=1, repeat=10))
+
+######################################################################
+# Sample Output
+# -------------
+# The following is the result of 'cpu', 'opencl' and 'vulkan' using Adreno 530 on Snapdragon 820
+#
+# .. code-block:: bash
+#
+#    Top-1 id: 282, class name: tiger cat
+#
+#    Top5 predictions: 
+#
+#         #1: tiger cat
+#         #2: Egyptian cat
+#         #3: tabby, tabby cat
+#         #4: lynx, catamount
+#         #5: red fox, Vulpes vulpes
+#          [282 285 281 287 277]
+#    Evaluate inference time cost...
+#    Execution time summary:
+#     mean (ms)   median (ms)    max (ms)     min (ms)     std (ms)  
+#      76.6101      76.6053      77.6530      75.5720       0.6789
```

Comment:
Did you try to convert this script into the web page?

Reply:
It must not be converted manually. It is done automatically when documentation is built

Reply:
Yes, I know, but probably it would be better to convert it locally and be sure that everything is fine in the final document.

Reply:
the main value is py script that is potentially can be executed by user. Why do we care about doc? It should be converted automatically...

Reply:
Because from this .py script, web page will be automatically generated and presented in the official TVM documentation, and I want to be sure that everything looks correctly. This is why I asked this question. I don't think that there is the problem to build this page locally and just check that everything is fine.